src/middleware/token-validation-middleware.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Middleware to handle the token authorization etc.
+ */
+
+import { Request, Response, NextFunction } from 'express';
+import jwt from 'jsonwebtoken';
+import { Core } from 'nodets-ms-core';
+import { environment } from "../environment/environment";
+import { PermissionRequest } from 'nodets-ms-core/lib/core/auth/model/permission_request';
+import { UnAuthenticated } from '../exceptions/http/http-exceptions';
+
+/**
+ * Validates the token and sends the user_id in the req.body
+ * the user id is available as `req.user_id`
+ * @param req - Initial request
+ * @param res  - Supposed response (to be filled by others)
+ * @param next - Next function
+ */
+export async function tokenValidator(req: Request, res: Response, next: NextFunction) {
+
+    const approvedRoles = ["tdei_admin", "poc", "flex_data_generator"]; // Change it based on service
```

Comment:
last one I think should be `pathways_data_generator`

Reply:
fixed